pkg/core/native/crypto.go

```diff
@@ -178,32 +178,11 @@ func curveFromStackitem(si stackitem.Item) (elliptic.Curve, error) {
 }
 
 func (c *Crypto) bls12381Serialize(_ *interop.Context, args []stackitem.Item) stackitem.Item {
-	val := args[0].(*stackitem.Interop).Value()
-	var res []byte
-	switch p := val.(type) {
-	case *bls12381.G1Affine:
-		compressed := p.Bytes()
-		res = compressed[:]
-	case *bls12381.G1Jac:
-		g1Affine := new(bls12381.G1Affine)
-		g1Affine.FromJacobian(p)
-		compressed := g1Affine.Bytes()
-		res = compressed[:]
-	case *bls12381.G2Affine:
-		compressed := p.Bytes()
-		res = compressed[:]
-	case *bls12381.G2Jac:
-		g2Affine := new(bls12381.G2Affine)
-		g2Affine.FromJacobian(p)
-		compressed := g2Affine.Bytes()
-		res = compressed[:]
-	case *bls12381.GT:
-		compressed := p.Bytes()
-		res = compressed[:]
-	default:
-		panic(errors.New("unknown bls12381 point type"))
+	val, ok := args[0].(*stackitem.Interop).Value().(blsPoint)
+	if !ok {
+		panic(errors.New("not a bls12381 point"))
 	}
-	return stackitem.NewByteArray(res)
+	return stackitem.NewByteArray(val.Bytes())
```

Comment:
And now I'm wondering: don't we want to move the rest of Groth-related logic (add, multiply, pairing, etc.) to `crypto_blspoints.go` and bind it to `blsPoint`?

Reply:
Well, a nice option too. At least we'd have more simple `crypto.go`.